string mag
---
int32 success